src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/MidEventCallback.java

```diff
@@ -62,21 +63,39 @@ public JsonNode invoke(String caseTypeId,
                 .findFirst();
 
             if (wizardPageOptional.isPresent() && !isBlank(wizardPageOptional.get().getCallBackURLMidEvent())) {
-                CaseDetails newCaseDetails = caseService.createNewCaseDetails(caseTypeId, caseType.getJurisdictionId(),
-                                                                              content.getEventData() == null ? content.getData() : content.getEventData());
+                Optional<CaseDetails> currentCaseDetails = populateCurrentCaseDetailsWithUserInputs(content, caseType);
+                CaseDetails newCaseDetails;
+                if (currentCaseDetails.isPresent()) {
+                    newCaseDetails = caseService.createNewCaseDetails(caseTypeId, caseType.getJurisdictionId(),
+                        currentCaseDetails.get().getData());
 
+                } else {
+                    newCaseDetails = caseService.createNewCaseDetails(caseTypeId, caseType.getJurisdictionId(),
+                        content.getEventData() == null ? content.getData() : content.getEventData());
+                }
                 CaseDetails caseDetails = callbackInvoker.invokeMidEventCallback(wizardPageOptional.get(),
-                                                                                 caseType,
-                                                                                 caseEvent,
-                                                                                 null,
-                                                                                 newCaseDetails,
-                                                                                 content.getIgnoreWarning());
+                    caseType,
+                    caseEvent,
+                    null,
+                    newCaseDetails,
+                    content.getIgnoreWarning());
                 return dataJsonNode(caseDetails.getData());
             }
         }
         return dataJsonNode(content.getData());
     }
 
+    private Optional<CaseDetails> populateCurrentCaseDetailsWithUserInputs(CaseDataContent content, CaseType caseType) {
```

Comment:
I was expecting the logic in this method to be part of CaseService. Looks more to be like responsibility of CaseService to merge and build CaseDetails as the other method createNewCaseDetails resides in CaseService as well. Do you think it will be good to refactor this a little bit?